mopidy/internal/storage.py

```diff
@@ -47,13 +47,15 @@ def dump(path, data):
     directory, basename = os.path.split(path)
 
     # TODO: cleanup directory/basename.* files.
-    tmp = tempfile.NamedTemporaryFile(
-        prefix=basename + '.', dir=directory, delete=False)
-
     try:
-        with gzip.GzipFile(fileobj=tmp, mode='wb') as fp:
-            json.dump(data, fp, cls=models.ModelJSONEncoder,
-                      indent=2, separators=(',', ': '))
+        with tempfile.NamedTemporaryFile(
+                prefix=basename + '.', dir=directory, delete=False) as tmp:
```

Comment:
I understand you had to rewrite this as `NamedTemporaryFile` cannot be opened twice on Windows. With this change, I believe you can simplify the code further:

1. Remove `delete=False` here, so we get the default `delete=True` behavior.
2. Remove the wrapping `try/finally` that is only there to clean up the temporary file. That is now handled by the context manager.